entities/auctioneer.py

```diff
@@ -19,65 +19,57 @@ def auction(self):
         # List of all buyers and sellers:
         print('+++ REGISTER +++')
         print('Buyers:')
-        print('       ID            Q     P')
-        print('----------------  ------ -----')
-        fmt = '{:7} {:8} {:5}'
+        print(' ID     P')
+        print('---- -----')
+        fmt = '{:4} {:4}'
         for buyer in self.buyers:
-            print(fmt.format(id(buyer), buyer.quantity, buyer.item_price))
+            print(fmt.format(buyer.agent_id, buyer.item_price))
 
         print('\nSellers:')
-        print('       ID            Q     P')
-        print('----------------  ------ -----')
-        for seller in self.sellers:
-            print(fmt.format(id(seller), seller.quantity, seller.item_price))
-
-
-        # AUCTION ALGORITHM
-        # (1) SELLERS: determine total supply and median item price:
-        total_q_s = 0
-        auction_items = []
+        print(' ID     P')
+        print('---- -----')
         for seller in self.sellers:
-            total_q_s += seller.quantity
-            for i in range(seller.quantity):
-                auction_items.append(seller.item_price)
-        median_p_s = median(sorted(auction_items))
+            print(fmt.format(seller.agent_id, seller.item_price))
 
-        # (2) BUYERS: determine total demand and median item price:
-        total_q_b = 0
-        auction_items_b = []
-        for buyer in self.buyers:
-            total_q_b += buyer.quantity
-            for j in range(buyer.quantity):
-                auction_items_b.append(buyer.item_price)
-        auction_items_b = sorted(auction_items_b, reverse=True)
-        median_p_b = median(auction_items_b[0:total_q_s])
-
-        # (3) Determine auction price:
-        auction_price = (median_p_s + median_p_b) / 2
+        # AUCTION ALGORITHM:
+        # For for every container an auction is run
+        while len(self.buyers) != 0:
+            for buyer in self.buyers:
+                for _ in range(100):
+                    if random() < 0.05:
+                        # Shipments (temporary: are refined later)
+                        shipment = Seller()
+                        shipment_id = id(shipment)
+                        self.shipments.append(shipment)
 
-        print('\n+++ AUCTION +++')
-        print("Total supply:        %s" % total_q_s)
-        print("Total demand:        %s" % total_q_b)
-        print("Median seller price:  %s" % median_p_s)
-        print("Median buyer price:   %s" % median_p_b)
-        print("Auction price:        %s" % auction_price)
+                winning_bid = max([shipment.item_price for shipment in self.shipments])
```

Comment:
`max(self.shipments, key=lambda item: item.item_price)` is better, as it does not recreate a list.

Reply:
(from experience: this should work, but I do not have access to a Python interpreter ATM)